eth/stagedsync/stage_polygon_sync.go

```diff
@@ -354,19 +354,19 @@ func (s polygonSyncStageCheckpointStore) GetLastEntityId(ctx context.Context) (u
 	return r.id, r.ok, r.err
 }
 
-func (s polygonSyncStageCheckpointStore) GetLastEntity(ctx context.Context) (cp *heimdall.Checkpoint, err error) {
+func (s polygonSyncStageCheckpointStore) GetLastEntity(ctx context.Context) (*heimdall.Checkpoint, bool, error) {
 	id, ok, err := s.GetLastEntityId(ctx)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 	if !ok {
-		return nil, errors.New("last checkpoint not found")
+		return nil, false, errors.New("last checkpoint not found")
```

Comment:
let's align the behaviour here with the behaviour in `mdbxEntityStore`
for the not found case we return `ok=false, err=nil`

Reply:
Did you mean `ok=true, err=nil`